### PR TITLE
Add dynamoQueryPagesCount metric

### PR DESCRIPTION
**Pull request by aaron7**
Metric for the number of pages we see per query inside QueryPages.

Test plan: `rate(cortex_dynamo_query_pages_count_sum[1m]) / rate(cortex_dynamo_query_pages_count_count[1m])` locally shows average page size of 1 which is expected.

### CODE REVIEW

File: pkg/chunk/aws_storage_client.go [hunk 1]
```diff
@@ -298,6 +306,7 @@ func (a awsStorageClient) QueryPages(ctx context.Context, query IndexQuery, call
 
 		backoff = minBackoff
 	}
+	dynamoQueryPagesCount.Observe(float64(pageCount))
```
Comment (jml) [inline pkg/chunk/aws_storage_client.go]: There are a couple of early returns in the loop, so this `Observe` call won't always happen.

**Comment by tomwilkie** [comment]
@aaron7 whilst this is a useful bit of instrumentation, if your dev-test loop include getting something reviewed and merged into master you're gonna have a bad time.   You should experiment with a custom built query service, deploy is straight to your dev environment (as a separate service perhaps) and then fire queries at it useing the socks proxy.  Then you'll be able to iterate and get to a solution much faster.

**Comment by aaron7** [comment]
>@aaron7 whilst this is a useful bit of instrumentation, if your dev-test loop include getting something reviewed and merged into master you're gonna have a bad time. You should experiment with a custom built query service, deploy is straight to your dev environment (as a separate service perhaps) and then fire queries at it useing the socks proxy. Then you'll be able to iterate and get to a solution much faster.

Agreed. I will look into setting up a way to deploy dev environments from local branches.

**Comment by tomwilkie** [comment]
Let me know if you need a hand.

**Pull request merged**